src/Business Foundation/App/NoSeriesCopilot/src/Copilot/NoSeriesCopilotImpl.Codeunit.al

```diff
@@ -0,0 +1,565 @@
+        Json.GetValueAndSetToRecFieldNo(RecRef, 'description', GeneratedNoSeries.FieldNo(Description));
+        Json.GetValueAndSetToRecFieldNo(RecRef, 'startingNo', GeneratedNoSeries.FieldNo("Starting No."));
+        Json.GetValueAndSetToRecFieldNo(RecRef, 'endingNo', GeneratedNoSeries.FieldNo("Ending No."));
+        Json.GetValueAndSetToRecFieldNo(RecRef, 'warningNo', GeneratedNoSeries.FieldNo("Warning No."));
+        Json.GetValueAndSetToRecFieldNo(RecRef, 'incrementByNo', GeneratedNoSeries.FieldNo("Increment-by No."));
+        Json.GetValueAndSetToRecFieldNo(RecRef, 'tableId', GeneratedNoSeries.FieldNo("Setup Table No."));
+        Json.GetValueAndSetToRecFieldNo(RecRef, 'fieldId', GeneratedNoSeries.FieldNo("Setup Field No."));
+        Json.GetValueAndSetToRecFieldNo(RecRef, 'nextYear', GeneratedNoSeries.FieldNo("Is Next Year"));
+        RecRef.Insert(true);
+
+        ValidateGeneratedNoSeries(RecRef);
+    end;
+
+    local procedure ValidateGeneratedNoSeries(var RecRef: RecordRef)
+    var
+        GeneratedNoSeries: Record "No. Series Proposal Line";
+    begin
+        ValidateRecFieldNo(RecRef, GeneratedNoSeries.FieldNo("Is Next Year"));
+        RecRef.Modify(true);
+    end;
+
+    local procedure ValidateRecFieldNo(var RecRef: RecordRef; FieldNo: Integer)
+    var
+        FieldRef: FieldRef;
+    begin
+        FieldRef := RecRef.Field(FieldNo);
+        FieldRef.Validate();
+    end;
+
+    local procedure SetProposalNo(var RecRef: RecordRef; GenerationId: Integer; FieldNo: Integer)
+    var
+        FieldRef: FieldRef;
+    begin
+        FieldRef := RecRef.Field(FieldNo);
+        FieldRef.Value(GenerationId);
+    end;
+
+    local procedure MinimumAccuracy(): Decimal
+    begin
+        exit(0.9);
+    end;
+
+    local procedure MaxInputTokens(): Integer
+    begin
+        exit(MaxModelTokens() - MaxOutputTokens());
+    end;
+
+    local procedure MaxOutputTokens(): Integer
+    begin
+        exit(4096);
+    end;
+
+    local procedure MaxModelTokens(): Integer
+    begin
+        exit(16385); //gpt-3.5-turbo-latest
+    end;
+
+    procedure IsCopilotVisible(): Boolean
+    var
+    // EnvironmentInformation: Codeunit "Environment Information";
+    begin
+        // if not EnvironmentInformation.IsSaaSInfrastructure() then //TODO: Check how to keep IsSaaSInfrastructure but be able to test in Docker Environment
+        //     exit(false);
+
+        if not IsSupportedLanguage() then
+            exit(false);
+
+        exit(true);
+    end;
+
+    local procedure IsSupportedLanguage(): Boolean
+    var
+        LanguageSelection: Record "Language Selection";
+        UserSessionSettings: SessionSettings;
+    begin
+        UserSessionSettings.Init();
+        LanguageSelection.SetLoadFields("Language Tag");
+        LanguageSelection.SetRange("Language ID", UserSessionSettings.LanguageId());
+        if LanguageSelection.FindFirst() then
+            if LanguageSelection."Language Tag".StartsWith('pt-') then
```

Comment:
@DenLilleMand We can remove this right? We no longer have the portugese limitation for any feature.

Reply:
@darjoo yes we should get rid of everything that filters Copilot features based on language or regions(e.g. Canada was denied access earlier)